src/compiler/program.ts

```diff
@@ -972,21 +959,93 @@ namespace ts {
         return resolutions;
     }
 
-    export function getDefaultTypeDirectiveNames(options: CompilerOptions, rootFiles: string[], host: CompilerHost): string[] {
+
+    /**
+      * Computes the types folder, a.k.a. the primary types lookup location.
+      * This is commonly './types' relative to whatever the compilation root is.
+      */
+    function getTypesFolderLocation(options: CompilerOptions, host: CompilerHost, inferredRoot: string): Path {
+        // Openish question: When tsconfig specifies a rootDir *and* a relative typesRoot, is the typesRoot
+        // relative to the tsconfig file path, or the rootDir path?
+
+        // If typesRoot came from tsconfig, then it's relative the config file path.
+        // Otherwise, it's relative to the current folder.
+        if (options.typesRoot) {
+            // true case is e.g. tsconfig { typesRoot: './foo' }, so types is [tsconfigPath]/foo
+            // false case is e.g. tsc a.ts --typesRoot foo, so types is [cwd]/foo

```

Comment:
We don't even have that information at this point; I'm inclined to say if you use `--p` *and* specify extra commandline values, the commandline values are still validly interpreted as being relative to the project file path